### PR TITLE
AP-5514: Added InvoiceReturn and specs

### PR DESCRIPTION
**Pull request by thomascrumrine**

### CODE REVIEW

File: spec/apruve/resources/invoice_return_spec.rb [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  it { should respond_to(:currency) }
+  it { should respond_to(:uuid) }
+  it { should respond_to(:reason) }
+  it { should respond_to(:merchant_notes) }
+  it { should respond_to(:created_by_id) }
+  it { should respond_to(:created_at) }
+  it { should respond_to(:updated_at) }
+
+  describe '#find' do
+    describe 'success' do
```
Comment (dbelling) [inline spec/apruve/resources/invoice_return_spec.rb]: Within a unit spec for a method, RSpec allows for [contexts](http://www.betterspecs.org/#contexts) to describe a particular scenario under scrutiny in the test. So for something like this:

```ruby
describe '#find'
  context 'a successful response'
    it 'should successfully get an invoice return'
      # Invoice Return expectations here
      ...
```

We've got several specs in our code that adhere to the pattern of nested describes, just my personal preference is to use contexts.

**Comment by dbelling** [comment]
One other quick thing - as part of this, or [#20](https://github.com/apruve/apruve-ruby/pull/20), you'll want to [bump](https://github.com/apruve/apruve-ruby/blob/master/lib/apruve/version.rb#L2) [the version](https://github.com/apruve/apruve-ruby/blob/master/apruve.gemspec#L8) on the gem and [publish a release on rubygems](http://guides.rubygems.org/publishing/#publishing-to-rubygemsorg). 

I don't believe we've typically synched library releases with app release cycles in the past, but I'll defer to Dave or Todd on that. The changes in [#20](https://github.com/apruve/apruve-ruby/pull/20) are necessary for some of the upcoming Quickbooks integration work.

**Comment by thomascrumrine** [comment]
Does this look better @dbelling ?

**Comment by donnen** [comment]
Looks good to me. We can publish a new version of the gem once we get the latest changes merged in. This can happen outside of the regular release cycle.

👍

**Pull request merged**